syft/frameworks/torch/linalg/operations.py

```diff
@@ -63,6 +65,192 @@ def _ldl(t):
+
+            # Send them to remote worker if t is pointer, secret share it if its an AST
+            if t_type == "pointer":
+                down_zeros = down_zeros.send(t.child.location)
+                up_zeros = up_zeros.send(t.child.location)
+            if t_type == "ast":
+                down_zeros = down_zeros.fix_prec(precision_fractional=prec_frac).share(
+                    *workers, crypto_provider=crypto_prov
+                )
+                up_zeros = up_zeros.fix_prec(precision_fractional=prec_frac).share(
+                    *workers, crypto_provider=crypto_prov
+                )
+            left_cat = torch.cat((I[:i, :i], down_zeros), dim=0)
+            right_cat = torch.cat((up_zeros, H), dim=0)
+            H = torch.cat((left_cat, right_cat), dim=1)
+
+        # Update R
+        R = H @ R
+        if not mode == "r":
+            # Update Q_transpose
+            Q_t = H @ Q_t
+
+    if mode == "reduced":
+        R = R[:n_cols, :]
+        Q_t = Q_t[:n_cols, :]
+
+    if mode == "r":
+        R = R[:n_cols, :]
+        return R
+
+    return Q_t.t(), R
+
+
+def _norm_mpc(t, norm_factor):
+    """
+    Computation of a norm of a vector in MPC. The vector should be an AdditiveSharedTensor.
+
+    It performs the norm calculation by masking the tensor with a multiplication
+    by a big random number drawn from a uniform distribution, and computing the
+    square root of the squared norm of the masked tensor, which is computed
+    beforehand with a dot product in MPC.
+
+    In order to maintain stability and avoid overflow, this functions uses a
+    norm_factor that scales the tensor for MPC computations and rescale it at the end.
+    For example in the case of the DASH algorithm, this norm_factor should be of
+    the order of the square root of number of entries in the original matrix
+    used to perform the compression phase assuming the entries are standardized.
+
+    Args:
+        t: 1-dim AdditiveSharedTensor, representing a vector.
+        norm_factor: float. The normalization factor used to avoid overflow
+
+    Returns:
+        q: orthogonal matrix as a 2-dim tensor with same type as t
+        r: lower triangular matrix as a 2-dim tensor with same type as t
+
+    """
+
+    workers = t.child.child.locations
+    crypto_prov = t.child.child.crypto_provider
+    prec_frac = t.child.precision_fractional
+    norm_factor = int(norm_factor)
+    t_normalized = t / norm_factor
+    Q = int(1e8)
+
+    norm_sq = (t_normalized * t_normalized).sum().squeeze()
+
+    # Random big number
+    r = (
+        torch.LongTensor([0])
+        .fix_precision(precision_fractional=prec_frac)
+        .share(*workers, crypto_provider=crypto_prov)
+        .random_(0, Q)
+    )
+
+    # Compute masked norm
+    masked_norm_sq = r ** 2 * norm_sq
+
+    # Get compute square root
+    masked_norm_sq = masked_norm_sq.get().float_precision()
+    masked_norm = torch.sqrt(masked_norm_sq)
```

Comment:
Can you send this stuff to be computed by the crypto provider instead of disclosing this to the local worker?

Reply:
I am not sure PySyft supports sending the shares to another remote worker, without calling `.get()` before. I tried to find a smart way to do it avoiding passing the shares to the local worker, but I didn't find a proper solution...

Do you have any ideas about how I can do that?